### PR TITLE
AddStandardWell functionality in OpenCL backend

### PR DESCRIPTION
**Pull request by ducbueno**
Clean PR, derived from https://github.com/OPM/opm-simulators/pull/2716

### CODE REVIEW

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
I will merge this as is if it does not break any other stuff after testing locally. The issues we are still facing can be resolved in separate PRs.

File: opm/simulators/linalg/bda/WellContributions.cpp [hunk 1]
```diff
@@ -24,79 +24,202 @@
...
+            delete[] val_pointers;
+        }
+    }
+#endif
+}
+
+#if HAVE_OPENCL
+
+void WellContributions::init(cl::Context *context){
+    d_Cnnzs_ocl = cl::Buffer(*context, CL_MEM_READ_WRITE, sizeof(double) * num_blocks * dim * dim_wells);
```
Comment (blattms) [inline opm/simulators/linalg/bda/WellContributions.cpp:107]: Sorry, cannot merge, yet. For opm-tests/model1/BASE1_MSW_HFA.DATA this throws an exception:
```
Error: [/home/mblatt/src/dune/opm-gpu/opm-simulators/opm/simulators/linalg/bda/openclSolverBackend.cpp:520] OpenCL Error: clCreateBuffer(-61)
CL_INVALID_BUFFER_SIZE
Program threw an exception: [/home/mblatt/src/dune/opm-gpu/opm-simulators/opm/simulators/linalg/bda/openclSolverBackend.cpp:520] OpenCL Error: clCreateBuffer(-61)
CL_INVALID_BUFFER_SIZE
```
num_blocks is 0 in this case. This model has no Standard wells, hence there should be no allocation here?
Please make sure that the above model works as before. There seems ti be more code that might falsely be executed if there are no standard wells (copyDataToGPU, init,

Comment (blattms) [reply opm/simulators/linalg/bda/WellContributions.cpp:107]: BTW the model runs with master.

Comment (ducbueno) [reply opm/simulators/linalg/bda/WellContributions.cpp:107]: I'll take a look

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
Merging this now to prevent more maintenance trouble with the opencl development. Simple models with only multisegment wells still run. (There is certainly a bug there as iteration counts fluctuate form run to run. These fluctuation already happened with master).

Please note that (at least on my system) standard wells still do not work with OpenCL. This needs to be fixed in subsequent PRs.

**Comment by blattms** [comment]
See #2754 for a problem to be resolved.

**Comment by blattms** [comment]
Thanks for the work so far.

**Pull request merged**